pkg/resolution/resolver/framework/fakeresolver.go

```diff
@@ -159,8 +167,12 @@ var _ TimedResolution = &FakeResolver{}
 
 // GetResolutionTimeout returns the configured timeout for the reconciler, or the default time.Duration if not configured.
 func (r *FakeResolver) GetResolutionTimeout(ctx context.Context, defaultTimeout time.Duration) time.Duration {
-	if r.Timeout > 0 {
-		return r.Timeout
+	return GetResolutionTimeout(r.Timeout, defaultTimeout)
+}
+
+func GetResolutionTimeout(timeout, defaultTimeout time.Duration) time.Duration {
```

Comment:
Shall this not be exported?

Reply:
It needs to be exported since it is used by the new package too.